plugins/module_utils/vmware.py

```diff
@@ -218,3 +224,141 @@ def is_vcenter(self):
             return True
         elif api_type == 'HostAgent':
             return False
+
+    def __get_objs_by_name(self, vimtype, name, first_only=False):
+        """
+        Get any vsphere object associated with a given text name and vim type.
+        Args:
+            vimtype: The type of object to search for
+            name: The name of the object to search for
+            first_only: If true, return once the first object is found.
+                        Useful when names must be unique
+        Returns:
+            list(object) or list() if no matches are found
+        """
+        objs = []
+        container = self.content.viewManager.CreateContainerView(self.content.rootFolder, vimtype, True)
+        for c in container.view:
+            if c.name == name:
+                if first_only:
+                    return c
+                objs += [c]
+
+        return objs if objs else None
+
+    def get_vm_by_name(self, vm_name, fail_on_missing=False, name_match=None):
```

Comment:
why not using the rest client for get VM?

Reply:
as far as i can tell, the rest client only allows you to search by name, not UUID or even MOID. if theres an example of that, it would be extremely helpful

Reply:
But the rest is only by name so no need this func